src/nupic/algorithms/Connections.hpp

```diff
@@ -41,15 +41,15 @@ namespace nupic
 
     namespace connections
     {
-      typedef UInt32 CellIdx;
-      typedef unsigned char SegmentIdx;
-      typedef unsigned char SynapseIdx;
+      typedef UInt CellIdx;
```

Comment:
The change to a short SegmentIdx is used in creating an invalid Segment (e.g. in bestMatchingCell and bestMatchingSegment). An alternative way could be implemented.